### PR TITLE
fix x86 cmp [mem], imm not getting data xrefs to mem

### PR DESCRIPTION
**Pull request by ulidtko**
Hey, I noticed that one specific instruction doesn't generate data xrefs as it should: `cmp [mem], imm`. The  `cmp reg, [mem]` form is fine, but with mem location on the left -- nope.

The fix seem to drop a stray line over there. No idea what it was supposed to do, it doesn't look intentional at all. Why assume that the immediate operand will be an address? Why the elaborate `op->ptr` calculations before? Why the difference between INSOP(0)/INSOP(1) cases?

Verified via `pij`, works more correctly now AFAICT.

### CODE REVIEW

**Comment by ulidtko** [comment]
Yea, some tests have failed because of changing `pd` output, like here:
```
Diff: --- /tmp/r2-regressions//cmd_types-r5i93R/exp	2018-10-25 10:24:10.318233664 +0000
+++ /tmp/r2-regressions//cmd_types-r5i93R/out	2018-10-25 10:24:10.346233707 +0000
@@ -2,4 +2,4 @@
             0x00000000      8b4004         mov eax, dword [eax + foo.b] ; [0x4:4]=0x508d0451
             0x00000003      895104         mov dword [ecx + foo.b], edx
             0x00000006      8d5008         lea edx, [eax + foo.c]      ; 0x8
-            0x00000009      83780800       cmp dword [eax + foo.c], 0
+            0x00000009      83780800       cmp dword [eax + foo.c], 0  ; [0x8:4]=0x8788308
[  ]  1  cmd_visual: visual intermediate    
```

Do I have an LGTM to try fixing the tests?

**Comment by radare** [comment]
yep

**Comment by radare** [comment]
ping

**Comment by ulidtko** [comment]
Hey, two questions here:

1. Test `db/anal/anal_tp` is obviously failing on type constraints deduction. I see this diff:

        This test has failed:
        File: ../bins/anal/constr_type
        Script: db/anal/anal_tp
        Name: constrained type
        Input: [ 'e anal.types.constraint = true',
          'aaa',
          's sym.single_cond',
          'pd 1~4h',
          's sym.range_small',
          'pd 1~14h',
          's sym.range_high',
          'pd 1~4h',
          's sym.range_or',
          'pd 1~4h' ]
        -|           ; var signed int local_4h  { > 0xa} @ rbp-0x4
        -|           ; var signed int local_14h  { > 0x0 && <= 0x9} @ rbp-0x14
        -|           ; var signed int local_4h  { > 0x64 && <= 0xc7} @ rbp-0x4
        -|           ; var signed int local_4h  { > 0xff && <= 0x12b || > 0x14 && <= 0x31 || > 0x6f && <= 0xdd} @ rbp-0x4
        +|           ; var signed int local_4h  { > 0xfffffffffffffffc} @ rbp-0x4
        +|           ; var signed int local_14h  { > 0xffffffffffffffec && <= 0xffffffffffffffec} @ rbp-0x14
        +|           ; var signed int local_4h  { > 0xfffffffffffffffc && <= 0xfffffffffffffffc} @ rbp-0x4
        +|           ; var signed int local_4h  { > 0xfffffffffffffffc && <= 0xfffffffffffffffc || > 0xfffffffffffffffc && <= 0xfffffffffffffffc || > 0xfffffffffffffffc && <= 0xfffffffffffffffc} @ rbp-0x4

    Indeed it looks like a legit regression. Doesn't appear in master, so must be that my change breaks it. But, I'm not really sure how to approach fixing this; marking broken for now.

    `db/cmd/cmd_search` also raises regression, again not sure how to fix, just marking it broken.

2. ~How do I submit the test fixes?~ Nevermind, I figured it out. Added r2r overlay. The commit amended to keep related changes together. Travis should in theory go green now (but +4 tests marked broken).

**Comment by Maijin** [comment]
@sivaramaaa can help for the constraints deductions

**Comment by ulidtko** [comment]
@Maijin `cmd_search` tests looks like they detect HTML content at every offset (0x000, 0x001, 0x002 and so on), any ideas with that?

File: libr/anal/p/anal_x86_cs.c [hunk 1]
```diff
@@ -2172,7 +2172,6 @@ static void anop(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len, csh
 				op->stackptr = regsz;
 				op->type |= R_ANAL_OP_TYPE_REG;
 			}
-			op->ptr = INSOP(1).imm;
```
Comment (sivaramaaa) [inline libr/anal/p/anal_x86_cs.c]: Yup this looks wrong , but the problem is there are lot of places where op->ptr and op->val is used interchangebly , let me try a proper fix for this !

**Comment by sivaramaaa** [comment]
I have added ur commit to my pr , so closing this !

**Pull request closed without merging**